### PR TITLE
BitArray: add opEquals(), and change allocation size to size_t

### PR DESCRIPTION
**Pull request by WalterBright**
It's inefficient to allocate these in anything other than size_t sizes.

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

File: src/dmd/root/bitarray.d [hunk 1]
```diff
@@ -55,6 +63,11 @@ nothrow:
             btc(ptr, idx);
     }
 
+    bool opEquals(const ref BitArray b) const
```
Comment (marler8997) [inline src/dmd/root/bitarray.d]: ddoc comment?

Comment (PetarKirov) [reply src/dmd/root/bitarray.d]: I agree, but given that Walter is unlikely to address this, I think that it's better to either push the change directly or open another PR

**Review by PetarKirov — APPROVED** [review]

File: src/dmd/root/bitarray.d (hunk 1 above)
Comment (WalterBright) [reply src/dmd/root/bitarray.d]: Being a standard function pattern, I didn't see a need.

Comment (marler8997) [reply src/dmd/root/bitarray.d]: Ok, thanks for the clarification on the standards

**Pull request merged**